deploy/helm/sumologic/conf/setup/resources.tf

```diff
@@ -29,13 +38,22 @@ resource "kubernetes_secret" "sumologic_collection_secret" {
   data = {
     {{- $ctx := .Values }}
     {{- range $type, $sources := .Values.sumologic.collector.sources }}
-    {{- if eq (include "terraform.sources.component_enabled" (dict "Context" $ctx "Type" $type)) "true" }}
-    {{- range $key, $source := $sources }}
-    {{- if eq (include "terraform.sources.to_create" (dict "Context" $ctx "Type" $type "Name" $key)) "true" }}
+      {{- if eq (include "terraform.sources.component_enabled" (dict "Context" $ctx "Type" $type)) "true" }}
+        {{- range $key, $source := $sources }}
+          {{- if eq (include "terraform.sources.to_create" (dict "Context" $ctx "Type" $type "Name" $key)) "true" }}
     {{ include "terraform.sources.data" (dict "Endpoint" (include "terraform.sources.config-map-variable" (dict "Type" $type "Context" $ctx "Name" $key)) "Name" (include "terraform.sources.name" (dict "Name" $key "Type" $type))) }}
-    {{- end }}
-    {{- end }}
-    {{- end }}
+          {{- end }}
+        {{- end }}
+      {{- else if and (eq $type "metrics") $ctx.sumologic.traces.enabled }}
+        {{- /*
+        If traces are enabled and metrics are disabled, create default metrics source anyway
+        */}}
+        {{- range $key, $source := $sources }}
+          {{- if eq $key "default"}}
+    {{ include "terraform.sources.data" (dict "Endpoint" (include "terraform.sources.config-map-variable" (dict "Type" $type "Context" $ctx "Name" $key)) "Name" (include "terraform.sources.name" (dict "Name" $key "Type" $type))) }}
```

Comment:
I think it would be more readable and less changes if we always create this endpoint (regardless of metrics and tracing configuration). If we have to add it for traces we can add it for logs as well and we just won't use it